internal/db/schema/edition.go

```diff
@@ -0,0 +1,56 @@
+package schema
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db/schema/internal/edition"
+)
+
+// Dialect is same as edition.Dialect
+type Dialect = edition.Dialect
+
+// Supported dialects.
+const (
+	Postgres Dialect = "postgres"
```

Comment:
In the internal `edition` package, there isn't really a reason to limit the dialect. If/when we support another dialect nothing would need to change in the edition package. However, the schema package does have a limited set of supported dialects. It will need to be updated to support those dialects. So adding the constrain here makes sense, but is extraneous in the `edition` package. It could just be a `string` in the `edition` package, but I think having a type makes the code more readable.

The alias cannot be reversed, since the `edition` package is an internal package and should not import `schema`, since `schema` needs to import `edition`.